atomix/cluster/src/main/java/io/atomix/raft/partition/RaftPartitionGroup.java

```diff
@@ -492,6 +492,21 @@ public Builder withPartitionDistributor(final PartitionDistributor partitionDist
       return this;
     }
 
+    /**
+     * Sets the threshold for preferring snapshot replication. When a follower lags behind by more
+     * than the threshold, the leader prefers sending a snapshot over replicating records.
+     *
+     * @param preferSnapshotReplicationThreshold the threshold to use
+     * @return this builder for chaining
+     */
```

Comment:
:wrench: I think when reading such a doc I would like to know what this threshold means, like the count of events. So if the follower lags X (< threshold) behind events are replicated reaching the threshold (count of events lagging behind) the snapshot is sent to the follower.

Reply:
How about

```java
  /**
   * Sets the threshold for preferring snapshot replication. The unit is <i>number of records</i>
   * by which a follower may lag behind before the leader starts to prefer replicating snapshots
   * instead of records.
   *
   * @param preferSnapshotReplicationThreshold the threshold to use
   * @return this builder for chaining
   */
```